### ISSUE TITLE
Spotify waybar problem

### ISSUE THREAD

**Issue by nickdritan** [prasanthrangan/hyprdots#508]
Whenever I play a track on spotify the bar updates with the new title, but the duration does not update. One more problem is that when I close spotify, the notification bar instead of showing nothing, it shows the last track track played. I found that if I run the script "wbarconfgen.sh" it does update it, but is there a way to do this automatically?

**Comment by T-Crypt** [comment]
@nickdritan 

I think we need to add the below to the spotify.jsonc module. 

Would you care testing this for me? 

`"escape": true,`

```
    "custom/spotify": {
        "exec": "/usr/bin/python3 $HOME/.config/waybar/modules/mediaplayer.py --player spotify",
        "format": "{} ",
        "return-type": "json",
        "on-click": "playerctl play-pause --player spotify",
        "on-scroll-up": "playerctl next",
        "on-scroll-down": "playerctl previous",
        "escape": true
    },

**Comment by T-Crypt** [comment]
@nickdritan 

How are you closing spotify? Are you using SUPER + Q? For me, when I close spotify with the super + q it removes the song title from the waybar. 

I think we need to add "escape": true to the spotify.jsonc in modules.

Can you test this by updating your spotify.jsonc in '~/.config/waybar/modules/spotify.jsonc` with the code below?

```
    "custom/spotify": {
        "exec": "/usr/bin/python3 $HOME/.config/waybar/modules/mediaplayer.py --player spotify",
        "format": "{} ",
        "return-type": "json",
        "on-click": "playerctl play-pause --player spotify",
        "on-scroll-up": "playerctl next",
        "on-scroll-down": "playerctl previous",
       "escape": true
    },

**Comment by nickdritan** [comment]
Hi, thanks for your response

Yes i close spotify with SUPER-Q or with middle-click in waybar,

I replaced the code  in '~/.config/waybar/modules/spotify.jsonc` with the one you provided , but the problem is still there

https://github.com/prasanthrangan/hyprdots/assets/113607157/5d94c2b3-b91e-4965-9318-b0ee3c31d9ca

![231117_19h23m06s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/113607157/da8404d8-24aa-4725-b9ed-e68b7bf7832e)

**Comment by T-Crypt** [comment]
@nickdritan okay, let me test!

**Comment by T-Crypt** [comment]
Okay so this is actually only for this waybar style -- it looks like its not using the spotify module and its usually the MPRIS module

**Comment by T-Crypt** [comment]
Can you update the mpris.jsonc module in the waybar/modules directory to add an interval to see if this resolves it? 

      "interval": 5

**Comment by nickdritan** [comment]
Updated it but it doesnt resolve it
![231117_20h26m42s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/113607157/31cbba52-7244-4d35-9390-92e4c05bab30)

**Comment by imhatep** [comment]
@nickdritan after saving the change, switch to the next one waybar (super+alt+up/down arrow) and come back, because reloading(ctrl+esc) it does not help

**Comment by nickdritan** [comment]
@imhatep You're right, now the track duration is working fine!
The only problem remaining is the title being there when I close spotify

**Comment by T-Crypt** [comment]
@imhatep @nickdritan 

Ill work on this and send a PR once I resolve this!

**Comment by T-Crypt** [comment]
Fixed in the latest PR #514 

Updated Module below -- Make sure to reload the config with SUPER + ALT + UP 

You can fix before the PR by replacing your mpris.jsonc module in the waybar directory!

```
  "mpris": {
        "format": "{player_icon} {dynamic}",
        "format-paused": "{status_icon} <i>{dynamic}</i>",
        "player-icons": { 
            "default": "▶",
            "mpv": "🎵"
        },
        "status-icons": {
            "paused": "⏸"
        },
        // "ignored-players": ["firefox"]
        "max-length": 1000,
        "interval": 3
    },
```

**Comment by nickdritan** [comment]
Thanks @T-Crypt , everything works now!
